tools/rosconsole/src/rosconsole/rosconsole.cpp

```diff
@@ -425,6 +435,25 @@ void initialize()
     backend::function_notifyLoggerLevelsChanged = notifyLoggerLevelsChanged;
     backend::function_print = _print;
 
+    char* line_buffered = NULL;
+#ifdef _MSC_VER
+    _dupenv_s(&line_buffered, NULL, "ROSCONSOLE_STDOUT_LINE_BUFFERED");
```

Comment:
In the Windows case you must free the allocated memory (`line_buffered`) afterwards.